aten/src/ATen/native/MaxUnpooling.cpp

```diff
@@ -460,13 +599,91 @@ Tensor max_unpooling2d_backward_cpu(
     const Tensor& grad_output,
     const Tensor& self,
     const Tensor& indices,
-    IntArrayRef output_size) {
+    IntArrayRef output_size,
+    IntArrayRef stride,
+    IntArrayRef padding) {
   auto grad_input = at::empty_like(self, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
   max_unpooling2d_backward_out_cpu(
-      grad_input, grad_output, self, indices, output_size);
+      grad_input, grad_output, self, indices, output_size, stride, padding);
+  return grad_input;
+}
+
+Tensor& max_unpooling1d_backward_out_cpu(
```

Comment:
This has the same issues with the forward. Namely, check shape before accessing elements and handle 2D vs 3D input size